### PR TITLE
410002 refactor audit work

### PR DESCRIPTION
**Pull request by feedmypixel**
## Refactor Audit work

I have been having a look at this a little over the last few weeks when I have had a little time and this is the result. I have added a ticket as the work got a little big. 

- Remove test work that was in production code, this is now done via
  Jest
- Pass request to the auditor so `cdpRequestId` is available to the Auditor, meaning we don't have to pass it in everywhere
- Remove pre work, this is now done in the Auditor
- Simplify config being passed to the plugin
- Update schema
- Pass an ISOString to the firehose data rather than a date object
- Update test coverage
- Simplify the `audit.send()` calls and make these calls async we don't need to wait around for a reply

## Dev logger

Screenshot 2024-07-09 at 18 00 00
Screenshot 2024-07-09 at 18 00 16

### CODE REVIEW

File: src/server/common/helpers/audit/auditor.js [hunk 1]
```diff
@@ -1,15 +1,48 @@
...
+    send: noop
+  }
+}
+
+const auditor = {
+  name: 'auditor',
+  register: async (
+    server,
+    {
+      audit,
```
Comment (flurdy) [inline src/server/common/helpers/audit/auditor.js:32]: where is `audit` from?

Comment (flurdy) [reply src/server/common/helpers/audit/auditor.js:32]: Ah is that the input. I see

File: src/server/index.js [hunk 2]
```diff
@@ -116,7 +116,10 @@ async function createServer() {
     nunjucksConfig,
     sanitise,
     router,
-    auditing
+    {
+      plugin: auditor,
+      options: { audit: config.get('audit') }
```
Comment (feedmypixel) [inline src/server/index.js:121]: @flurdy audit is being sent here via options

File: src/server/common/helpers/audit/auditor.js (hunk 1 above)
Comment (feedmypixel) [reply src/server/common/helpers/audit/auditor.js:32]: Over [here](https://github.com/DEFRA/cdp-portal-frontend/pull/460/files#r1670901646)

**Review by flurdy — APPROVED** [review]

File: src/server/index.js (hunk 2 above)
Comment (flurdy) [reply src/server/index.js:121]: Cool

**Pull request merged**